### PR TITLE
Add attributes_to_get parameter to Model.scan

### PR DESCRIPTION
**Pull request by ButtaKnife**
Add the attributes_to_get parameter to Model.scan to enable projection expressions on scan operations.

### CODE REVIEW

**Comment by akashg90** [comment]
Is there any plan to release this change?

**Comment by ikonst** [comment]
@garrettheel lgtm, agree?

**Comment by garrettheel** [comment]
Ideally we'd add support for ProjectionExpressions, as AttributesToGet is now deprecated, but this is fine in the interim.

Can you please rebase and also update the stubs in the `.pyi` files (`indexes.pyi` and `models.pyi`)? [[example](https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/models.pyi#L96)]

File: .coveragerc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [run]
 source = pynamodb/.
-omit = pynamodb/tests/integration/*
\ No newline at end of file
+omit = pynamodb/tests/*
```
Comment (ikonst) [inline .coveragerc]: wait, what? :)

File: .travis.yml [hunk 2]
```diff
@@ -3,12 +3,17 @@ language: python
 python:
   - "3.6"
   - "3.5"
-  - "3.4"
-  - "3.3"
   - "2.7"
-  - "2.6"
```
Comment (ikonst) [inline .travis.yml]: ?!

**Comment by nvbillb** [comment]
Oh hey, I had not seen the most recent replies on this PR since it's so ancient.  I was just pulling in the latest changes, but I'll do a rebase and see about addressing the stubs.  It would be great if this could be merged soon.

Also, @garrettheel - agreed about ProjectionExpressions vs AttributesToGet, but I wasn't quite comfortable enough with the PynamoDB codebase to make that type of refactor yet.

**Comment by ButtaKnife** [comment]
(Posting this time with the correct account...)
Oh hey, I had not seen the most recent replies on this PR since it's so ancient. I was just pulling in the latest changes, but I'll do a rebase and see about addressing the stubs. It would be great if this could be merged soon.

Also, @garrettheel - agreed about ProjectionExpressions vs AttributesToGet, but I wasn't quite comfortable enough with the PynamoDB codebase to make that type of refactor yet.

File: .coveragerc (hunk 1 above)
Comment (ButtaKnife) [reply .coveragerc]: Something with the merge commit from before was not visualizing correctly here.  I did a rebase instead now and things are much cleaner.

File: .travis.yml (hunk 2 above)
Comment (ButtaKnife) [reply .travis.yml]: Same for this - rebase cleaned this right up.

**Comment by garrettheel** [comment]
Sorry for the delay @ButtaKnife. Would you mind resolving the conflicts and pinging me again so we can get this in?

**Comment by ButtaKnife** [comment]
@garrettheel - Merged the latest!  We've been using these changes against a fairly recent version of master in microservices at work and it's been good so far.  Would be amazing to get the PR merged in here so we can stop using my fork.  :)

**Comment by garrettheel** [comment]
Thanks again!

**Pull request merged**